R/auth.R

```diff
@@ -1,74 +1,121 @@
-                      client_secret = NULL,
-                      token = NULL,
-                      ...) {
-  # client id or secret not provided
-  if ((is.null(client_id) | is.null(client_secret)) & is.null(token)) {
-    stop("Please generate a client secret and client key following these instructions:\n",
-         "https://www.hairizuan.com/rgoogleslides-using-your-own-account-client-id-and-secret/")
+authorize <- function(token = NULL, cache = FALSE, ...) {
+  if (!cache) {
+    cache_it <- menu(c("Yes store credentials as .httr-oauth file", "No do not store credentials, I will re-run this authorize() in my next R session"))
+    if (cache_it == 1) {
+      message("You chose to cache your credentials, if you change your mind, just delete the .httr-oauth. Be careful not to push this file to GitHub or share it anywhere.")
+    }
+  } else {
+    cache_it <- 1
   }
   if (is.null(token)) {
-    # setup app
-    app <- httr::oauth_app(appname = "googleslides",
-                           key = client_id,
-                           secret = client_secret)
-    # google endpoints
-    endpoint <- httr::oauth_endpoints("google")
-    # generate token
-    token <- httr::oauth2.0_token(endpoint = endpoint,
-                                  app = app,
-                                  scope = c("https://www.googleapis.com/auth/presentations",
-                                            "https://www.googleapis.com/auth/drive.readonly"),
-                                  ...)
+    token <- httr::oauth2.0_token(
+      endpoint = app_set_up()$endpoint,
+      app = app_set_up()$app,
+      cache = cache_it == 1,
+      scope = scopes_list,
+      ...
+    )
   }
   set_token(token)
-  invisible(token)
+  return(invisible(token))
+}
+
+#' Use secrets to authorize R package to access Google Slides API
+#' @description This is a function to authorize the R package to access the Google Slides API. If no
+#' client.id and client.secret is provided, the package would provide predefined values.
+#' @param access_token Access token can be obtained from running authorize() interactively: token <-authorize(); token$credentials$access_token
+#' @param refresh_token Refresh token can be obtained from running authorize() interactively: token <-authorize(); token$credentials$refresh_token
+#' @return OAuth token saved to the environment so the package can use the users' Google data
+#' @importFrom utils menu installed.packages
+#' @importFrom httr oauth_app oauth_endpoints oauth2.0_token
+#' @export
+#' @examples \dontrun{
+#'
+#' token <- authorize()
+#'
+#' auth_from_secret(
+#'   token$credentials$access_token,
+#'   token$credentials$refresh_token
+#' )
+#' }
+#'
+auth_from_secret <- function(access_token = NULL, refresh_token = NULL) {
+  if (is.null(access_token) | is.null(refresh_token)) {
+    decrypted <- openssl::aes_cbc_decrypt(
+      readRDS(encrypt_creds_user_path()),
+      key = readRDS(key_encrypt_creds_path())
+    )
```

Comment:
```suggestion
    )
access_token <- unserialize(decrypted)[[1]]$access_token
refresh_token <- unserialize(decrypted)[[1]]$refresh_token
```
I'm doing this on my phone but this is what I'm suggesting you do. Bring the phrases from below up here (not sure if I typed these right) and declare the variables. Then below just specify `access_token = access_token` and same for refresh_token.

If this doesn't make sense, let me know and I'll send a commit later.